modules/cas_cache/layer_cache_management.c

```diff
@@ -2385,11 +2397,31 @@ int cache_mngt_init_instance(struct ocf_mngt_cache_config *cfg,
 	switch (cmd->init_cache) {
 	case CACHE_INIT_LOAD:
 	case CACHE_INIT_STANDBY_LOAD:
-		result = _cache_mngt_check_metadata(cfg, cmd->cache_path_name);
+		result = _cache_mngt_probe_metadata(cmd->cache_path_name,
+				cache_name_meta, &cache_mode_meta,
+				&cache_line_size_meta);
 		if (result) {
 			module_put(THIS_MODULE);
 			return result;
 		}
+
+		result = ocf_mngt_cache_get_by_name(cas_ctx, cache_name_meta,
+				OCF_CACHE_NAME_SIZE, &tmp_cache);
+		/* Need to return name from metadata now for caller to properly
+		 * communicate the error to user */
+		BUG_ON(cache_id_from_name(&cmd->cache_id, cache_name_meta));
```

Comment:
I would move cache_id_from_name() call to line 2407 - now it is located strangely right between setting the result and checking it.

Reply:
RE BUG_ON: I think that it's apropriate - OCF returned no error, so I'm assuming all data retrieved from metadata is correct, if it's not then we're in trouble

Reply:
With this BUG_ON it's possible to prepare metadata on drive (actually just change cache name and recalculate CRC, so nothing sophisticated) so that loading in causes kernel panic. In case of cache name it's even worse, because malicious intervention is not even needed - we have OCF in SPDK, and it can produce metadata with cache names not compatible with OCL convention. Please add proper error handling.

Reply:
That's right, I've fixed it and moved the call earlier.